hbase-common/src/main/java/org/apache/hadoop/hbase/TagBuilder.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright The Apache Software Foundation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase;
+
+import java.nio.ByteBuffer;
+import org.apache.yetus.audience.InterfaceAudience;
+
+/**
+ * Builder method to create {@link Tag}.
+ */
+@InterfaceAudience.LimitedPrivate(HBaseInterfaceAudience.COPROC)
+public interface TagBuilder {
+  /**
+   * Set type of the tag.
+   * @param tagType type of the tag
+   * @return {@link TagBuilder}
+   */
+  TagBuilder setTagType(byte tagType);
+
+  /**
+   * For {@link ArrayBackedTag}, set the value of the tag.
+   * This method throws {@link UnsupportedOperationException}
+   * if the tag is backed by {@link ByteBuffer}
+   * @param tagBytes tag bytes.
+   * @return {@link TagBuilder}
+   */
+  TagBuilder setTagValue(byte[] tagBytes);
```

Comment:
When create a new Tag (Not reading HFile data and creating back Tag POJO), actually it does not matter much whether we pass value as byte[] or BB.  Infact as a user I would have created array backed value and so array backed tag only.
We can give both options anyways in Builder. But we can have it as overloaded method? 
setTagValue (or setValue is enough as we are anyways in tagBuilder)
setTagValue(byte[])
setTagValue(BB)
In impl the latest set one will anyways be used.

Reply:
Thank you for the review. Addressed it in latest commit.